spec/cypress/integration/cip-stories-early-career-teacher.js

```diff
@@ -0,0 +1,55 @@
+describe("ECT user interaction with Core Induction Programme", () => {
+  beforeEach(() => {
+    cy.login("early_career_teacher");
+  });
+
+  it("should not show a download export button", () => {
+    cy.visit("/core-induction-programme");
+    cy.contains("a.govuk-button", "Download export").should("not.exist");
+  });
+
+  it("should not allow to edit year title", () => {
+    cy.appFactories([["create", "course_year"]]);
+
+    cy.visitYear();
+    cy.contains("a.govuk-button", "Edit year content").should("not.exist");
+  });
+
+  it("should not allow to edit module title", () => {
+    cy.appFactories([["create", "course_module"]]);
+
+    cy.visitModule();
+    cy.contains("a.govuk-button", "Edit module content").should("not.exist");
+  });
+
+  it("should not allow to edit lesson title", () => {
+    cy.appFactories([["create", "course_lesson"]]);
+
+    cy.visitLesson();
+    cy.contains("a.govuk-button", "Edit lesson").should("not.exist");
+  });
+
+  it("should not allow to edit lesson part title", () => {
+    cy.appFactories([["create", "course_lesson", "with_lesson_part"]]);
+
+    cy.visitLesson();
+    cy.get("h2").should("contain", "Title");
+    cy.contains("a.govuk-button", "Edit lesson content").should("not.exist");
+  });
+
+  it("should display lesson progress", () => {
+    cy.appFactories([["create", "course_lesson", "with_lesson_part"]]);
+
+    cy.visitModule();
+    cy.contains("strong.govuk-tag", "not started");
+
+    cy.visitLesson();
+    cy.visitModule();
+    cy.contains("strong.govuk-tag", "in progress");
+
+    cy.visitLesson();
+    cy.get('[type="checkbox"]').check();
+    cy.get("form").submit();
+    cy.contains("strong.govuk-tag", "complete");
+  });
```

Comment:
sorry for not thinking about this in my first review, but the way the helpers you've created work and are being used is a bit of an anti-pattern. e.g., with this test you are doing:

1. visit year
2. [assertion]
3. visit year
4. visit module
5. visit lesson
6. visit year
7. visit module
8. [assertion]
9. visit year
10. visit module
11. visit lesson
12. [assertions]

Also, if you create more than one lesson/module, the helper will fail, as it won't know which link to click on.

I can't immediately think of an actually good solution, but one thought I've had is that `visitLesson()` could behave the same way as `login()` does - it could create a lesson, and then immediately navigate to the lesson it just created using the URL. Then any navigating between levels (e.g. from lesson to module) could be done without a helper, by clicking on the back link.

Reply:
I pushed an update - feel free to have a look :)